Source/OIDAuthState.m

```diff
@@ -462,35 +462,40 @@ - (void)performActionWithFreshTokens:(OIDAuthStateAction)action
     // creates a list of pending actions, starting with this one
     _pendingActions = [NSMutableArray arrayWithObject:action];
   }
+    
+    __weak typeof(self) weakSelf = self;
 
   // refresh the tokens
   OIDTokenRequest *tokenRefreshRequest =
       [self tokenRefreshRequestWithAdditionalParameters:additionalParameters];
   [OIDAuthorizationService performTokenRequest:tokenRefreshRequest
                                       callback:^(OIDTokenResponse *_Nullable response,
                                                  NSError *_Nullable error) {
+                                          
+    __strong typeof(weakSelf) strongSelf = weakSelf;
+                                          
     dispatch_async(dispatch_get_main_queue(), ^() {
       // update OIDAuthState based on response
       if (response) {
```

Comment:
Can we add a `if (strongSelf != nil) {return;}` here?  We recently got bit on that one.